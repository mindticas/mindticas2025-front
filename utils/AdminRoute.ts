import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
