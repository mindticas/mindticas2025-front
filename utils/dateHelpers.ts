// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}