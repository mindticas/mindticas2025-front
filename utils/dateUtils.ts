import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
