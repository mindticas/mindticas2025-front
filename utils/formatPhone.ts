export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
