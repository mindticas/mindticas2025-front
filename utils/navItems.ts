import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
