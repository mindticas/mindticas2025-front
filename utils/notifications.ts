import { toaster } from '@/components/ui/toaster';

type ToastType = 'success' | 'error' | 'info' | 'warning';

interface ToastOption {
    duration?: number;
    title: string;
    description?: string;
}

const showToast = (type: ToastType, options: ToastOption) => {
    toaster.create({
        type,
        duration: options.duration || 5000,
        title: options.title,
        description: options.description,
    });
};

const TreatmentNotification = {
    createSuccess: () =>
        showToast('success', { title: 'Tratamiento creado con éxito' }),
    updateSuccess: () =>
        showToast('success', { title: 'Tratamiento actualizado con éxito' }),
    deleteSuccess: () =>
        showToast('success', { title: 'Tratamiento eliminado con éxito' }),
    createError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al crear el tratamiento',
        }),
    updateError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al actualizar el tratamiento0',
        }),
    deleteError: () =>
        showToast('error', { title: 'Error al eliminar el tratamiento' }),
    noChanges: () =>
        showToast('info', {
            title: 'No se detectaron cambios para actualizar',
        }),
};

export default TreatmentNotification;
