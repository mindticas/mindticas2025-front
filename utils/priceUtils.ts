const parseTreatmentPrice = (price: unknown): number => {
    if (typeof price === 'number') return price;
    if (typeof price === 'string') {
        const numericValue = parseFloat(price.replace(/[^0-9.-]/g, ''));
        return isNaN(numericValue) ? 0 : numericValue;
    }
    return 0;
};

export default parseTreatmentPrice;
