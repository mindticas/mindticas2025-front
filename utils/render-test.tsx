import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
