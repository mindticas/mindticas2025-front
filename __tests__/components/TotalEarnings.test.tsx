import { render } from '@/utils/render-test';

// Primero definimos todos los datos mock que serán usados
const MOCK_DATA = {
    statistics: {
        totalEarnings: 300,
        totalServices: 2,
        totalCompletedAppointments: 2,
        totalCanceledAppointments: 0,
    },
    dateRange: {
        startDate: '2024-05-01T00:00:00Z',
        endDate: '2024-05-10T00:00:00Z',
    },
    appointments: [
        {
            scheduled_start: '2024-05-01T10:00:00Z',
            status: 'completed',
            treatments: [{ name: 'Corte', price: '100' }],
        },
        {
            scheduled_start: '2024-05-02T11:00:00Z',
            status: 'completed',
            treatments: [{ name: 'Tinte', price: '200' }],
        },
    ],
    treatments: [
        { id: '1', name: 'Corte', price: '100' },
        { id: '2', name: 'Tinte', price: '200' },
    ],
};

// Mock de servicios usando una función factory
jest.mock('@/services/AppointmentService', () => ({
    getAppointments: jest.fn(() => Promise.resolve(MOCK_DATA.appointments)),
}));

// Mock de hooks
jest.mock('@/hooks/useTreatments', () => ({
    useTreatments: () => ({
        treatments: MOCK_DATA.treatments,
        error: null,
    }),
}));

beforeEach(() => {
    jest.clearAllMocks();
});

describe('TotalEarnings Component', () => {
    it('renders correctly with statistics and date range', async () => {
        const { asFragment, findByText } = render(
            <TotalEarnings
                statistics={MOCK_DATA.statistics}
                dateRange={MOCK_DATA.dateRange}
            />,
        );

        await findByText(/ganancias por tratamiento/i);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders message when no dateRange is provided', () => {
        const { getByText } = render(
            <TotalEarnings statistics={undefined} dateRange={undefined} />,
        );

        expect(getByText(/selecciona un rango de fechas/i)).toBeInTheDocument();
    });
});
