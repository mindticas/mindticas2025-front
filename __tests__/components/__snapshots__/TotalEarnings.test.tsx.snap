// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalEarnings Component renders correctly with statistics and date range 1`] = `<DocumentFragment />`;
